h2/src/test/org/h2/test/db/TestFunctions.java

```diff
@@ -2543,4 +2544,47 @@ public void init(Connection conn) {
         // ignore
     }
 
+    private void testTranslateDB2() throws SQLException {
+        Connection conn = getConnection("functions;MODE=DB2");
+        Statement stat = conn.createStatement();
+
+        String createSQL = "CREATE TABLE testTranslate(id BIGINT, " +
+                "txt1 varchar);";
+        stat.execute(createSQL);
+        stat.execute("insert into testTranslate(id, txt1) " +
+                "values(1, 'test1')");
+        stat.execute("insert into testTranslate(id, txt1) " +
+                "values(2, null)");
+        stat.execute("insert into testTranslate(id, txt1) " +
+                "values(3, '')");
+        stat.execute("insert into testTranslate(id, txt1) " +
+                "values(4, 'caps')");
+
+        String query = "SELECT translate(txt1, 'p', 'r') " +
```

Comment:
for DB2 this should be 

    SELECT translate(txt1, 'r', 'p') 

surely?

Reply:
Please, swap arguments in your test case. You actually used Oracle's / H2's syntax in DB2 compatibility mode.